chaincode/permissions.go

```diff
@@ -53,25 +53,23 @@ func (perms Permissions) CanProcess(owner, node string) bool {
 
 // NewPermissions create the Permissions according to the arg received
 func NewPermissions(db *LedgerDB, in inputPermissions) (Permissions, error) {
-	nodes, err := queryNodes(db, []string{})
-	if err != nil {
-		return Permissions{}, err
-	}
-
-	nodesIDs := []string{}
-	for _, node := range nodes {
-		nodesIDs = append(nodesIDs, node.ID)
-	}
+	if !in.Process.Public {
```

Comment:
What would you think of the following to reduce indentation

```
if in.Process.Public {
  return
}
```

Reply:
Hmm we would still need to create the Permission object to assign proper owner and process/download perms.
That would mean moving this part before the validation, which I'd rather not do since this is a loaded footgun to bypass the validation.
I will extract this check in another function to cleanly separate the concerns.

Reply:
@mblottiere Oh my bad, I misread the function, your initial implementation is great.